### PR TITLE
[ADD][9.0] hr_timesheet_sheet_week_start_day

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
This module allows to define the week day that the employee timesheet should
 start on.
# Configuration

To configure this module, you need to:
# . Go to 'Settings / Companies / Companies' and then to tab 'Configuration'
# . Set the 'Timesheet range' to "Week"
# . Change the "Week start day" to the desired day of the week
# Usage

When an employee creates a new weekly timesheet, the default starting day
will be the week day defined in the company settings.

### CODE REVIEW

File: hr_timesheet_sheet_week_start_day/README.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
+.. image:: https://img.shields.io/badge/license-AGPLv3-blue.svg
+   :target: https://www.gnu.org/licenses/agpl.html
+   :alt: License: AGPL-3
+
+=================================
+Hr Timesheet Sheet Week Start Day
```
Comment (elicoidal) [inline hr_timesheet_sheet_week_start_day/README.rst:6]: HR weekly timesheets sheet start day?

File: hr_timesheet_sheet_week_start_day/README.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/hr-timesheet/issues>`_. In case of trouble, please
+check there if your issue has already been reported. If you spotted it first,
+help us smashing it by providing a detailed and welcomed `feedback
+<https://github.com/OCA/
```
Comment (elicoidal) [inline hr_timesheet_sheet_week_start_day/README.rst]: remove this part following newest template

File: hr_timesheet_sheet_week_start_day/README.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+------------
+
+* Eficent Business and IT Consulting Services S.L. <contact@eficent.com>
+
+Maintainer
+----------
+
+.. image:: https://odoo-community.org/logo.png
+   :alt: Odoo Community Association
+   :target: http://odoo-community.org
```
Comment (elicoidal) [inline hr_timesheet_sheet_week_start_day/README.rst]: https (x2)

File: hr_timesheet_sheet_week_start_day/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# © 2015 Eficent Business and IT Consulting Services S.L. (www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Hr Timesheet Sheet Week Start Day",
```
Comment (elicoidal) [inline hr_timesheet_sheet_week_start_day/__openerp__.py:6]: check README

File: hr_timesheet_sheet_week_start_day/tests/test_hr_timesheet_sheet_week_start_day.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    def test_create_timesheet(self):
+        hrtss = self.hrtss_model.create({'employee_id': self.employee.id,
+                                         'company_id': self.company.id})
+        date_from = fields.Date.from_string(hrtss.date_from)
+        date_to = fields.Date.from_string(hrtss.date_to)
+        weekday_from = date_from.weekday()
+        weekday_to = date_to.weekday()
+
+        self.assertEqual(weekday_from, 6, "The timesheet should start on "
+                                          "sunday")
```
Comment (elicoidal) [inline hr_timesheet_sheet_week_start_day/tests/test_hr_timesheet_sheet_week_start_day.py]: Sunday

File: hr_timesheet_sheet_week_start_day/tests/test_hr_timesheet_sheet_week_start_day.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+        self.assertEqual(weekday_to, 5, "The timesheet should end on saturday")
```
Comment (elicoidal) [inline hr_timesheet_sheet_week_start_day/tests/test_hr_timesheet_sheet_week_start_day.py]: Saturday

**Comment by elicoidal** [comment]
some details but :+1: 
Thanks for the contribution!

**Comment by elicoidal** [comment]
some details but :+1: 
Tested in runbot OK
Thanks for the contribution!

**Comment by rafaelbn** [comment]
Tested un runbot :+1:

File: hr_timesheet_sheet_week_start_day/README.rst (hunk 1 above)
Comment (JordiBForgeFlow) [inline hr_timesheet_sheet_week_start_day/README.rst:6]: I think is better to keep the name of the module in line with the technical name.

**Comment by JordiBForgeFlow** [comment]
@dufresnedavid @raycarnes @max3903  this one may be interesting to you. Your feedback is welcome!

File: hr_timesheet_sheet_week_start_day/README.rst (hunk 1 above)
Comment (elicoidal) [inline hr_timesheet_sheet_week_start_day/README.rst:6]: agree: if you change one, change the other

Comment (JordiBForgeFlow) [inline hr_timesheet_sheet_week_start_day/README.rst:6]: do you need me to change anything? Now the technical name is hr_timesheet_sheet_week_start_day, and the module name in openerp.py and README.rst is "Hr Timesheet Sheet Week Start Day".

Comment (elicoidal) [inline hr_timesheet_sheet_week_start_day/README.rst:6]: My title was a suggestion as current name is a little bit awkward, that's all.
Module names is fine to me.
(non blocking)

File: hr_timesheet_sheet_week_start_day/models/hr_timesheet_sheet.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    @api.model
+    def _default_date_from(self):
+        date_from = super(HrTimesheetSheet, self)._default_date_from()
+        user = self.env['res.users'].browse(self.env.uid)
+        r = user.company_id and user.company_id.timesheet_range or 'month'
+        if r == 'week':
+            if user.company_id.timesheet_week_start:
+                date_from = (datetime.today() + relativedelta(
+                    weekday=int(user.company_id.timesheet_week_start),
+                    days=-6)).strftime('%Y-%m-%d')
```
Comment (ghost) [inline hr_timesheet_sheet_week_start_day/models/hr_timesheet_sheet.py]: please use fields.Date.tostring

File: hr_timesheet_sheet_week_start_day/models/hr_timesheet_sheet.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        if r == 'week':
+            date_to = (datetime.today() + relativedelta(
+                weekday=week_end)).strftime('%Y-%m-%d')
+
+        return date_to
+
+    date_from = fields.Date(string='Date from', required=True, select=1,
+                            readonly=True,
+                            states={'new': [('readonly', False)]},
+                            default=_default_date_from)
```
Comment (ghost) [inline hr_timesheet_sheet_week_start_day/models/hr_timesheet_sheet.py]: if it's only to change the default, you don't have to rewrite the field entirely

date_from = fields.Date(default=_default_date_from)

File: hr_timesheet_sheet_week_start_day/models/hr_timesheet_sheet.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+            if user.company_id.timesheet_week_start:
+                date_from = (datetime.today() + relativedelta(
+                    weekday=int(user.company_id.timesheet_week_start),
+                    days=-6)).strftime('%Y-%m-%d')
+        return date_from
+
+    @api.model
+    def _default_date_to(self):
+        date_to = super(HrTimesheetSheet, self)._default_date_to()
+        user = self.env['res.users'].browse(self.env.uid)
```
Comment (ghost) [inline hr_timesheet_sheet_week_start_day/models/hr_timesheet_sheet.py]: user = self.env.user

File: hr_timesheet_sheet_week_start_day/models/hr_timesheet_sheet.py [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+        r = user.company_id and user.company_id.timesheet_range or 'month'
```
Comment (ghost) [inline hr_timesheet_sheet_week_start_day/models/hr_timesheet_sheet.py]: with the new Odoo api "user.company_id and" is redondant

r = user.company_id.timesheet_range or 'month'

File: hr_timesheet_sheet_week_start_day/views/res_company_view.xml [hunk 11]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data>
+        <record id="hr_timesheet_sheet_company" model="ir.ui.view">
+            <field name="name">res.company.sheet</field>
+            <field name="model">res.company</field>
+            <field name="inherit_id"
+                   ref="hr_timesheet_sheet.hr_timesheet_sheet_company"/>
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='timesheet_range']"
```
Comment (ghost) [inline hr_timesheet_sheet_week_start_day/views/res_company_view.xml]: you can simplify that to

`<field name="timesheet_range" position="after">`

File: hr_timesheet_sheet_week_start_day/views/res_company_view.xml [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+                       position="after">
+                        <field name="timesheet_week_start"
+                               attrs="{'invisible':[('timesheet_range','!=','week')]}"/>
```
Comment (ghost) [inline hr_timesheet_sheet_week_start_day/views/res_company_view.xml]: maybe add

'required':[('timesheet_range','=','week')]

**Comment by ghost** [comment]
@jbeficent thanks for the contribution. This module is indeed relevent to contribute. Few technical comments.

**Comment by ghost** [comment]
@jbeficent that was fast. thanks :+1:

**Comment by JordiBForgeFlow** [comment]
Thanks @dufresnedavid! This one is ready to merge.

**Comment by JordiBForgeFlow** [comment]
Sorry @dufresnedavid. One more small glitch found here: https://github.com/OCA/hr-timesheet/pull/80

**Comment by tafaRU** [comment]
@jbeficent thanks a lot for the module!
I see version number and copyright year should be changed respectively to 9.0.1.0.0 and to 2016.
If you like it tomorrow morning I'll provide a PR in that way. Let me know. Thanks.

**Pull request merged**